pkgs/applications/emulators/citra/generic.nix

```diff
@@ -0,0 +1,119 @@
+{ pname
+, version
+, src
+, branch
+, compat-list
+
+, lib
+, stdenv
+, config
+, fetchFromGitHub
+, catch2
+, cmake
+, ninja
+, pkg-config
+, wrapQtAppsHook
+, qtbase
+, qtdeclarative
+, qtmultimedia
+, qttools
+, boost17x
+, ffmpeg
+, libusb1
+, rapidjson
+, SDL2
+, zstd
+, libressl
+, alsaSupport      ? true, alsa-lib
+, pulseSupport     ? true, libpulseaudio
+, udevSupport      ? true, udev
+, sndioSupport     ? true, sndio
+, fdkSupport       ? true, fdk_aac
+, jackaudioSupport ? true, libjack2
+, discordSupport   ? true
+, onlineSupport    ? true
+}:
+
+stdenv.mkDerivation rec {
+  inherit pname version src;
+
+  cmakeFlags = [
+    "-DBUILD_SHARED_LIBS=OFF"
+    "-DCITRA_ENABLE_COMPATIBILITY_REPORTING=ON"
+    "-DCITRA_USE_BUNDLED_FFMPEG=OFF"
+    "-DCITRA_USE_BUNDLED_QT=OFF"
+    "-DCITRA_USE_BUNDLED_SDL2=OFF"
+    "-DDYNARMIC_ENABLE_CPU_FEATURE_DETECTION=OFF"
+    "-DENABLE_COMPATIBILITY_LIST_DOWNLOAD=OFF" # Sacrifices reproducibility
+    "-DENABLE_FFMPEG_VIDEO_DUMPER=ON"
+    "-DENABLE_QT_TRANSLATION=ON"
+    "-DUSE_SYSTEM_BOOST=ON"
+    "-GNinja"
+    "-Wno-dev"
+    "-DUSE_DISCORD_PRESENCE=${if discordSupport then "ON" else "OFF"}"
+    "-DENABLE_WEB_SERVICE=${if onlineSupport then "ON" else "OFF"}"
+    (if fdkSupport then "-DENABLE_FDK=ON" else "-DENABLE_FFMPEG_AUDIO_DECODER=ON")
+  ];
+
+  nativeBuildInputs = [
+    catch2
+    cmake
+    ninja
+    pkg-config
+    qttools
+    wrapQtAppsHook
+  ];
+
+  buildInputs = [
+    boost17x
+    ffmpeg
+    libusb1
+    qtbase
+    qtdeclarative
+    qtmultimedia
+    rapidjson
+    SDL2
+  ] ++ lib.optional alsaSupport      alsa-lib
+    ++ lib.optional fdkSupport       fdk_aac
+    ++ lib.optional jackaudioSupport libjack2
+    ++ lib.optional pulseSupport     libpulseaudio
+    ++ lib.optional sndioSupport     sndio
+    ++ lib.optional udevSupport      udev;
+
+  postPatch = ''
+    # Prep compatibilitylist
+    ln -s ${compat-list} ./dist/compatibility_list/compatibility_list.json
+
+    # Devendor some submodules
+    rm -rf externals/{libressl,zstd}
+    tar -xf ${libressl.src} --strip-components 1 --one-top-level="externals/libressl"
+    cp -r ${zstd.src} ./externals/zstd
+  '';
+
+  preConfigure = ''
+    # Trick the configure system. CMakeLists.txt#L55 function(check_submodules_present)
+    chmod 775 -R ./externals
```

Comment:
I think yuzu has a way nicer fix for this, we should make it possible to do the same here and upstream that some day.